09-spring-boot-ehcache-xml/src/main/java/com/example/springbootehcache/ext/ehcache/EhcachejCacheSpringConfiguration.java

```diff
@@ -0,0 +1,110 @@
+package com.example.springbootehcache.ext.ehcache;
+
+import org.ehcache.core.spi.service.InstantiatorService;
+import org.ehcache.spi.service.Service;
+import org.ehcache.spi.service.ServiceProvider;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanClassLoaderAware;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.boot.autoconfigure.cache.CacheProperties;
+import org.springframework.boot.autoconfigure.cache.JCacheManagerCustomizer;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.Resource;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
+
+import javax.cache.CacheManager;
+import javax.cache.Caching;
+import javax.cache.configuration.MutableConfiguration;
+import javax.cache.spi.CachingProvider;
+import java.io.IOException;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Properties;
+
+;
+
+@Configuration(proxyBeanMethods = false)
+@ConditionalOnClass({org.ehcache.CacheManager.class})
+@ConditionalOnMissingBean(org.springframework.cache.CacheManager.class)
+@ConditionalOnProperty("spring.cache.jcache.config")
+public class EhcachejCacheSpringConfiguration implements BeanClassLoaderAware, BeanFactoryAware, InstantiatorService {
```

Comment:
this class could go in a `ehcache-spring-boot` jar to auto activate Ehcache JCache integration with Ehcache services wired from spring beans